src/Treeview.ts

```diff
@@ -297,6 +292,40 @@ export class JsonOutlineProvider implements vscode.TreeDataProvider<string> {
         }
         return;
     }
+
+    private updateTreeState() {
+        const activeEditor: vscode.TextEditor = vscode.window.activeTextEditor;
+        const document: vscode.TextDocument = !!activeEditor ? activeEditor.document : null;
+        this.parseTree(document);
+        const showTreeView = this.isArmTemplate(document);
+
+        if (showTreeView) {
+            this.refresh();
+        }
+
+        this.setTreeViewContext(showTreeView);
+    }
+
+    private isArmTemplate(document?: vscode.TextDocument): boolean {
+        return !!document && document.languageId.toLowerCase() === 'json' && Utilities.isValidSchemaUri(this.getSchemaUri());
```

Comment:
Does getting the schema uri have any performance impact on large json files? In other words - does it parse the whole file and would the user notice any delay in their non-ARM json files because of the ARM extension?

Reply:
At this point the whole file has already been parsed. Getting the schema doesn't do a re-parse. There are room for perf improvements, but that will require much changes. I'd prefer not to do it at this time.